### PR TITLE
ApcNet updating fix

### LINKED ISSUE #3056: ApcNets do not update without an Apc
They need to have their own system instead of being updated by `PowerApcSystem.cs`.

### PR DESCRIPTION
**Pull request by collinlunn**
Closes #3056

Moves updating of ApcNets to a grid component, since there is no individual component that guarantees the existence of an ApcNet that I can do an EntityQuery for.

~TODO: Ensure ApcNets always stop getting updated after removal.~

### CODE REVIEW

File: Content.IntegrationTests/Tests/PowerTest.cs [hunk 1]
```diff
@@ -226,8 +226,10 @@ public async Task ApcNetTest()
             {
                 var mapMan = IoCManager.Resolve<IMapManager>();
                 var entityMan = IoCManager.Resolve<IEntityManager>();
-                mapMan.CreateMap(new MapId(1));
+                var mapId = new MapId(1);
+                mapMan.CreateMap(mapId);
                 var grid = mapMan.CreateGrid(new MapId(1));
```
Comment (metalgearsloth) [inline Content.IntegrationTests/Tests/PowerTest.cs:231]:
```suggestion
                var grid = mapMan.CreateGrid(mapId);
```

File: Content.IntegrationTests/Tests/PowerTest.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿using System.Threading.Tasks;
+using System.Threading.Tasks;
```
Comment (metalgearsloth) [inline Content.IntegrationTests/Tests/PowerTest.cs:1]:
```suggestion
#nullable enable
using System.Threading.Tasks;
```

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
1 minor thing and 1 nullable. Rest LGTM.

**Comment by collinlunn** [comment]
Will take some time to fix the test to be compatible with nullable.

**Review by PaulRitter — CHANGES_REQUESTED** [review]
you have nullability errors

**Comment by metalgearsloth** [comment]
If it's painful to fix the test just out of scope it.

**Comment by Zumorica** [comment]
Any reason this is a grid component instead of an entity system? This seems like it'd make creating "proper" new grids with power support more difficult

**Comment by collinlunn** [comment]
Fixing nullability on the power test is big enough to be its own PR, undoing that for now.

**Comment by collinlunn** [comment]
> Any reason this is a grid component instead of an entity system? This seems like it'd make creating "proper" new grids with power support more difficult

@Zumorica 
I could make the entity system maintain the set of ApcNets to update, but I'm not sure how I'd efficiently check which ApcNets are paused other than checking if each ApcNet's associated grid is paused every frame. If that's not a problem, I could move it to the entity system though.

**Comment by Zumorica** [comment]
Hmm, it is true that making this a grid component elegantly solves the paused grid problem. I *guess* newly created APCs could ensure that there's a GridPower component on the grid they're on...

**Comment by metalgearsloth** [comment]
> 
> 
> > Any reason this is a grid component instead of an entity system? This seems like it'd make creating "proper" new grids with power support more difficult
> 
> @Zumorica
> I could make the entity system maintain the set of ApcNets to update, but I'm not sure how I'd efficiently check which ApcNets are paused other than checking if each ApcNet's associated grid is paused every frame. If that's not a problem, I could move it to the entity system though.

You could check if the APC itself is paused during the uniqueApcNet update though checking it at a grid-level is faster.

**Comment by Zumorica** [comment]
Maybe APC nets could store which grid ID they belong to? That'd allow you to check if the grid is paused before updating, I guess.

**Comment by collinlunn** [comment]
> Maybe APC nets could store which grid ID they belong to? That'd allow you to check if the grid is paused before updating, I guess.

I could have an entity system maintain a set of all ApcNets, and do that. I just wasn't sure if it was efficient to check GridPaused so often, but that might be negilgible?

**Comment by Zumorica** [comment]
> > Maybe APC nets could store which grid ID they belong to? That'd allow you to check if the grid is paused before updating, I guess.
> 
> I could have an entity system maintain a set of all ApcNets, and do that. I just wasn't sure if it was efficient to check GridPaused so often, but that might be negilgible?

Sounds good imo.

**Comment by collinlunn** [comment]
I've updated this to follow the feedback, the grid component has been replaced with each ApcNet adding itself to an EntitySystem.

File: Content.Server/GameObjects/EntitySystems/ApcNetSystem.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        [Dependency] private readonly IPauseManager _pauseManager = default!;
+
+        private HashSet<IApcNet> _apcNets = new();
+
+        public override void Update(float frameTime)
+        {
+            foreach (var apcNet in _apcNets)
+            {
+                var gridId = apcNet.GridId;
+                if (gridId != null && !_pauseManager.IsGridPaused((GridId) gridId))
```
Comment (metalgearsloth) [inline Content.Server/GameObjects/EntitySystems/ApcNetSystem.cs:24]:
```suggestion
                if (gridId != null && !_pauseManager.IsGridPaused(gridId.Value))
```

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**